test/integration_test.go

```diff
@@ -4980,9 +4980,14 @@ func (ts *IntegrationTestSuite) TestScheduleUpdate() {
 		err = handle.Delete(ctx)
 		ts.NoError(err)
 	}()
+
+	stringKey := temporal.NewSearchAttributeKeyString("CustomStringField")
+	sa := temporal.NewSearchAttributes(stringKey.ValueSet("CustomStringFieldValue"))
+
 	updateFunc := func(input client.ScheduleUpdateInput) (*client.ScheduleUpdate, error) {
 		return &client.ScheduleUpdate{
-			Schedule: &input.Description.Schedule,
+			Schedule:         &input.Description.Schedule,
+			SearchAttributes: &sa,
```

Comment:
is there also a test (somewhere else in the file?) that an update without SearchAttributes set doesn't modify the search attributes?

Reply:
I agree the test should be added. I'd like to better understand the testing responsibilities for SDK/CLI and OSS. Do we generally expect all contract-related test cases to reside in both places? Or maybe edge case contract tests in OSS and happy path in CLI/SDK?

Reply:
I think there's a test in the server for a missing SearchAttributes (at the grpc level) to update the schedule with no change to SAs. What I'd like to test here is that leaving out SearchAttributes in ScheduleUpdate actually sends a grpc call with a missing SearchAttributes. I'd be happy with a unit test that does that, but it might be easier to do an integration test since it's already set up here.

Reply:
In general the SDK APIs should have tests for all of their documented behaviour. What the server tests at the grpc level doesn't validate the API at the SDK level.

Reply:
Added tests